### DISCUSSION TITLE
Configuring pylsp's auto-import to omit "src" from import statement

### DISCUSSION THREAD

**Discussion by daria-shaw** [python-lsp/python-lsp-server#525] [Q&A]
Hi all,

I'm having issues configuring pylsp to not add "src" when importing code from within my repository.

e.g. I'm working on a REPO using poetry and want to import a class y from a directory in `src/`

what I expect:
```python
from REPO.x import y
```

what I get:
```python
from src.REPO.x import y
```

I'm using pylsp within neovim (here is my [config](https://github.com/daara-s/dotfiles/blob/main/private_dot_config/nvim/lua/core/lsp_config.lua#L44-L57) for setting it up)

I've tried setting the `python_path` based on the usage note in the [docs](https://github.com/python-lsp/python-lsp-server/blob/develop/docs/autoimport.md)
```lua
rope = {python_path = {"src"}}   -- also tried with src/
```
but it's still suggesting the import as `from src.REPO.x import y`.
I've also ensured that neovim is running within the poetry environment.

Does anyone know how to set up pylsp's auto-import completion without it using `src.` in the import statement?

**Comment by ccordoba12** [comment]
@tkrabel, what do you think about this one?

**Comment by tkrabel** [comment]
@daara-s just a QQ: does that import completion lead to a syntax error or is it working, but not what you want?

**Reply by daria-shaw** [reply]
The `src.REPO` import completion is valid syntax and works fine, but formatting-wise for the repo it should be without `src.`

**Comment by daria-shaw** [comment]
I'll try adding the `src/` folder to a global [pytool.toml](https://rope.readthedocs.io/en/latest/configuration.html#pytool-toml) config `source_folders` tomorrow at work and see if that covers the problem.

**Reply by daria-shaw** [reply]
No luck with setting `python_files` or `source_folders` to variations of `["src"]` with `/` and `/*`. 
Still comes out with `src.REPO`

**Reply by tkrabel** [reply]
I think the settings don't do what they think you do, but I am also not 100% sure about them :)

After 10 min of reading rope code, my understanding is this: 

For features like rope_autoimport, rope creates a sqlite database which contains the object name and the module path from where to import it ([src](https://github.com/python-rope/rope/blob/master/rope/contrib/autoimport/models.py#L112-L120)). To create the sqlite DB, rope scans all python packages under `sys.path` ([src](https://github.com/python-rope/rope/blob/master/rope/contrib/autoimport/sqlite.py#L390)), and I didn't find any pointer in the code suggesting that you can change the import paths in the way would like to (i.e., not adding the "src." in the import path; @bagel897 and @lieryan keep me honest here). 

If my understanding is correct, I'd suggest you close this issue here and open an issue on [the repo](https://github.com/python-rope/rope) to request such a configurability, referencing #525 .

**Reply by bagel897** [reply]
There's an option to add more paths but I forgot which one it is. Rope uses the sys path from the venv it is in, which isn't ideal when it's different from the project you're working on.

**Reply by daria-shaw** [reply]
Thanks for helping to figure out a solution, I've made a feature request for rope: https://github.com/python-rope/rope/issues/753
